docs/storage_transformers/sharding/v1.0.rst

```diff
@@ -0,0 +1,191 @@
+  as restricted by the file block size and maximum inode number for typical
+  file systems.
+
+This does not necessarily fit the access patterns of the data, so chunks might
+need to be smaller than one storage key. In those cases sharding decouples those
+entities. One shard corresponds to one storage key, but can contain multiple chunks:
+
+.. image:: sharding.png
+
+
+Document conventions
+====================
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
+Configuration
+=============
+
+Sharding can be configured per array in the :ref:`array-metadata`:
+
+.. code-block::
+
+    {
+      storage_transformers: [
+        {
+          "storage_transformer": "https://purl.org/zarr/spec/storage_transformers/sharding/1.0",
+          "configuration": {
+            "format": "indexed",
+            "chunks_per_shard": [
+                2,
+                2
+            ]
+          }
+      ]
+    }
+
+``format``
+
+    Specifies a `Binary shard format`_. In this version, the only binary format is the
+    ``indexed`` format.
+
+``chunks_per_shard``
+
+    An array of integers providing the number of chunks that are combined in a shard
+    for each dimension of the Zarr array, where each chunk may only start at a position
+    that is divisble by ``chunks_per_shard`` per dimension, e.g. starting at the zero-origin.
+    The length of the array must match the length of the array metadata ``shape`` entry.
+    For example, a value ``[32, 2]`` indicates that 64 chunks are combined in one shard,
+    32 along the first dimension, and for each of those 2 along the second dimension.
+    Valid starting positions for a shard in the chunk-grid are therefore ``[0, 0]``,
+    ``[32, 2]``, ``[32, 4]``, ``[64, 2]`` or ``[96, 18]``.
+
+
+Storage transformer implementation
+==================================
+
+Key & value transformation
+--------------------------
+
+The storage transformer protocol defines the abstract interface to be the same
+as the :ref:`abstract-store-interface`.
+
+The Zarr store interface is defined in terms of `keys` and `values`,
+where a `key` is a sequence of characters and a `value` is a sequence
+of bytes. A key-value pair is called `entry` in the following part.
+
+This sharding transformer only adapts entries where the key starts
+with `data/root`, as they indicate data keys for array chunks. All other
```

Comment:
where does that come from? is that a constant or an example? can this be referenced to another part of the spec?

Reply:
That's part of the storage-key spec, a constant. I added a reference